src/Authentication/Authentication/custom/common/GraphCommand.ps1

```diff
@@ -31,4 +31,36 @@ function GraphCommand_ReadGraphCommandMetadata {
         $FileProvider.Dispose()
     }
     return $Result
+}
+
+function GraphCommand_ReadGraphCommandMapping {
+    [CmdletBinding()]
+    param(
+        [Parameter(Position = 0)]
+        [string]$Path = (Join-Path $PSScriptRoot "MgCommandMapping.json")
+    )
+    if (!(Test-Path $Path)) {
+        throw "MgCommandMapping file not found at $Path."
+    }
+
+    ## The following command will produce new json content for "MgCommandMapping.json" containing the latest cmdlet mappings from documentation.
+    # Invoke-WebRequest 'https://learn.microsoft.com/en-us/powershell/microsoftgraph/azuread-msoline-cmdlet-map?view=graph-powershell-1.0' | Select-Object -ExpandProperty Content | Select-String -Pattern '<td>.*?([a-z0-9-]+).*?</td>\s?<td><a href=.*>(.*)</a></td>' -AllMatches | Select-Object -ExpandProperty Matches | ForEach-Object { @{ LegacyMapping = $_.Groups[1].Value; Command = $_.Groups[2].Value } } | Group-Object Command | ForEach-Object { [pscustomobject][ordered]@{ Command = $_.Name; LegacyMapping = [string[]]$_.Group.LegacyMapping } } | ConvertTo-Json
```

Comment:
These legacy module commands are no longer being updated and changes to the graph powershell command names should be rare as well so theoretically this list is fairly static.

However, it would be great to have a better way of getting this mapping.
Just in case, I added an option to parse the raw documentation markdown as well which is probably (only slightly) more resilient. :)

Reply:
I agree with you @jasoth! I think we can merge this for now since we have most of the legacy commands in the current snapshot. We can change the legacy mapping source in future releases.